lib/js/components/Switch/Switch.vue

```diff
@@ -0,0 +1,336 @@
+					:icon="iconRight"
+					:size="ICON_SIZES.XX_SMALL"
+				/>
+				<div v-if="labelRight" class="switch__label">
+					{{ labelRight }}
+				</div>
+			</div>
+			<div
+				class="switch__item -selection"
+				:class="{
+					'-left': currentSide === SWITCH_SIDE.LEFT,
+					'-right': currentSide === SWITCH_SIDE.RIGHT,
+				}"
+			>
+				<ds-icon
+					v-if="currentIcon"
+					class="switch__icon"
+					:icon="currentIcon"
+					:size="ICON_SIZES.XX_SMALL"
+				/>
+				<div v-if="currentLabel" class="switch__label">{{ currentLabel }}</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../styles/settings/animations';
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/radiuses';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/typography/tokens';
+
+$switch-transition: all $default-transition-time ease-out;
+
+.switchWrapper {
+	display: inline-flex;
+	max-width: 100%;
+	overflow: hidden;
+}
+
+.switch {
+	$root: &;
+
+	align-items: flex-start;
+	background-color: $color-default-background;
+	border: 1px solid $color-neutral-border;
+	border-radius: $radius-xs;
+	box-sizing: border-box;
+	display: inline-flex;
+	max-width: 100%;
+	position: relative;
+
+	&.-rounded {
+		border-radius: $radius-xl;
+	}
+
+	&__icon {
+		color: $color-neutral-icon;
+	}
+
+	.-selected &__icon {
+		color: $color-primary-icon;
+	}
+
+	&__label {
+		@include label-s-extensive-bold-uppercase;
+
+		color: $color-neutral-text;
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+	}
+
+	.-selected &__label {
+		color: $color-neutral-text-heavy;
+	}
+
+	&__item {
+		align-items: center;
+		background-color: $color-neutral-background-ghost;
+		border: 1px solid transparent;
+		border-radius: $radius-xs;
+		cursor: pointer;
+		display: flex;
+		flex-grow: 1;
+		gap: $space-xxxs;
+		justify-content: center;
+		margin: -1px;
+		min-height: 32px;
+		min-width: 64px;
+		overflow: hidden;
+		padding: 0 $space-s;
+		position: relative;
+		transition: $switch-transition;
+		z-index: 2;
+
+		&.-left {
+			margin: -1px 0 -1px -1px;
+		}
+
+		&.-right {
+			margin: -1px -1px -1px 0;
+		}
+
+		&.-selection {
+			background-color: $color-primary-background;
+			border: 1px solid $color-primary-border;
+			box-sizing: border-box;
+			left: 0;
+			position: absolute;
+			top: 0;
+			z-index: 1;
+
+			#{$root}__icon {
+				visibility: hidden;
+			}
+
+			#{$root}__label {
+				overflow: hidden;
+				visibility: hidden;
+				white-space: nowrap;
+				width: max-content;
+			}
+
+			&.-left {
+				left: 0;
+			}
+
+			&.-right {
+				left: calc(100% + 1px);
+				transform: translateX(-100%);
+			}
+		}
+
+		&.-selected,
+		&.-selection:hover {
+			cursor: default;
+			flex-shrink: 0;
+			overflow: hidden;
+			text-overflow: initial;
+		}
+
+		&.-clickable:hover {
+			background-color: $color-neutral-background-ghost-hovered;
+		}
+	}
+
+	&.-small &__item {
```

Comment:
Maybe it should be an `__item` internal modifier class since it only modifies its elements

Reply:
Then it needs to be defined in 3 places: `.item.-left`, `.item.-right` and `.selection`. I wasn't sure if it's better to avoid duplication or simplify the css selector here. I'm fine with both approaches.

Reply:
Ok. So leave it here then